toolchains/BUILD

```diff
@@ -7,3 +27,465 @@ filegroup(
+)
+
+java_import(
+    name = "TestRunner",
+    jars = ["@remote_java_tools//:Runner"],
+)
+
+alias(
+    name = "TestRunner_deploy.jar",
+    actual = "@remote_java_tools//:Runner",
+)
+
+alias(
+    name = "proguard",
+    actual = "@remote_java_tools//:proguard",
+)
+
+# TODO(cushon): this isn't compatible with JDK 9
+alias(
+    name = "bootclasspath",
+    actual = "@local_jdk//:bootclasspath",
+)
+
+alias(
+    name = "jre",
+    actual = "@local_jdk//:jre",
+)
+
+alias(
+    name = "jdk",
+    actual = "@local_jdk//:jdk",
+)
+
+alias(
+    name = "host_jdk",
+    actual = ":remote_jdk11",
+)
+
+bootclasspath(
+    name = "platformclasspath",
+    src = "DumpPlatformClassPath.java",
+    host_javabase = "current_java_runtime",
+    target_javabase = "current_java_runtime",
+)
+
+default_java_toolchain(
+    name = "toolchain",
+    configuration = DEFAULT_TOOLCHAIN_CONFIGURATION,
+    toolchain_definition = False,
+)
+
+alias(
+    name = "remote_toolchain",
+    actual = ":toolchain",
+)
+
+RELEASES = (8, 9, 10, 11)
+
+[
+    default_java_toolchain(
+        name = "toolchain_java%d" % release,
+        configuration = DEFAULT_TOOLCHAIN_CONFIGURATION,
+        source_version = "%s" % release,
+        target_version = "%s" % release,
+    )
+    for release in RELEASES
+]
+
+# A toolchain that targets java 15.
+default_java_toolchain(
+    name = "toolchain_jdk_15",
+    configuration = dict(),
+    java_runtime = "//toolchains:remotejdk_15",
+    source_version = "15",
+    target_version = "15",
+)
+
+# A toolchain that targets java 16.
+default_java_toolchain(
+    name = "toolchain_jdk_16",
+    configuration = dict(),
+    java_runtime = "//toolchains:remotejdk_16",
+    source_version = "16",
+    target_version = "16",
+)
+
+# A toolchain that targets java 17.
+default_java_toolchain(
+    name = "toolchain_jdk_17",
+    configuration = dict(),
+    java_runtime = "//toolchains:remotejdk_17",
+    source_version = "17",
+    target_version = "17",
+)
+
+# Deprecated, do not use.
+# It will be removed after migration to Java toolchain resolution.
+default_java_toolchain(
+    name = "toolchain_hostjdk8",
+    configuration = JVM8_TOOLCHAIN_CONFIGURATION,
+    java_runtime = ":current_host_java_runtime",
+    source_version = "8",
+    target_version = "8",
+    toolchain_definition = False,
+)
+
+default_java_toolchain(
+    name = "prebuilt_toolchain",
+    configuration = PREBUILT_TOOLCHAIN_CONFIGURATION,
+    toolchain_definition = False,
+)
+
+filegroup(
+    name = "bzl_srcs",
+    srcs = glob(["*.bzl"]),
+    visibility = ["//tools:__pkg__"],
+)
+
+py_binary(
+    name = "proguard_whitelister",
+    srcs = [
+        "proguard_whitelister.py",
+    ],
+    deps = [
+        "//third_party/py/abseil",
+    ],
+)
+
+py_test(
+    name = "proguard_whitelister_test",
+    srcs = ["proguard_whitelister_test.py"],
+    data = ["proguard_whitelister_test_input.pgcfg"],
+    deps = [
+        ":proguard_whitelister",
+    ],
+)
```

Comment:
Remove, you don't have the source files here. (Also remove py_binary and py_test import).

Reply:
Done